x-pack/legacy/plugins/actions/server/lib/execute.ts

```diff
@@ -25,27 +30,32 @@ export async function execute({
   params,
   encryptedSavedObjectsPlugin,
 }: ExecuteOptions): Promise<ActionTypeExecutorResult> {
-  // TODO: Ensure user can read the action before processing
-  const action = await encryptedSavedObjectsPlugin.getDecryptedAsInternalUser('action', actionId, {
+  // Ensure user can read the action before processing
+  const {
+    attributes: { actionTypeId, config, description },
+  } = await services.savedObjectsClient.get<RawAction>('action', actionId);
+  // Only get encrypted attributes here, the remaining attributes can be fetched in
```

Comment:
Not immediately clear why this is getting everything but secrets in one call, secrets in another, instead of one call to get all of them.

Reply:
I was thinking of two scenarios. The first one is we need to make sure the user has access to the action (at least `read` access) by the time it executes. The second is in the future if Kibana supports attribute based access control (which probably wouldn't make sense in action scenario).

@kobelb should we just get all the attributes from the `getDecryptedAsInternalUser`? And then only make sure savedObjectsClient doesn't throw on get?

Reply:
I missed that the second call was with `...AsInternalUser()`, and so doesn't do the security check ...